syncmaster/worker/controller.py

```diff
@@ -243,21 +246,38 @@ def _perform_incremental_transfer(self) -> None:
         ).force_create_namespace() as hwm_store:
 
             with IncrementalStrategy():
-                if self.source_handler.connection_dto.type in FILE_CONNECTION_TYPES:
-                    hwm_name_suffix = self.source_handler.transfer_dto.directory_path
-                else:
-                    hwm_name_suffix = self.source_handler.transfer_dto.table_name
-                hwm_name = "_".join(
-                    [
-                        str(self.source_handler.transfer_dto.id),
-                        self.source_handler.connection_dto.type,
-                        hwm_name_suffix,
-                    ],
-                )
+                hwm_name = self._get_transfer_hwm_name()
                 hwm = hwm_store.get_hwm(hwm_name)
 
                 self.source_handler.hwm = hwm
                 self.target_handler.hwm = hwm
 
                 df = self.source_handler.read()
                 self.target_handler.write(df)
+
+    def _get_transfer_hwm_name(self) -> str:
+        if self.source_handler.connection_dto.type in FILE_CONNECTION_TYPES:
+            hwm_name_suffix = self.source_handler.transfer_dto.directory_path
+        else:
+            hwm_name_suffix = self.source_handler.transfer_dto.table_name
+        hwm_name = "_".join(
+            [
+                str(self.source_handler.transfer_dto.id),
+                self.source_handler.connection_dto.type,
+                hwm_name_suffix,
+            ],
+        )
+        return hwm_name
+
+    def _reset_transfer_hwm(self) -> None:
+        with HorizonHWMStore(
+            api_url=self.settings.hwm_store.url,
+            auth=LoginPassword(login=self.settings.hwm_store.user, password=self.settings.hwm_store.password),
+            namespace=self.settings.hwm_store.namespace,
+        ).force_create_namespace() as hwm_store:
+
+            hwm_name = self._get_transfer_hwm_name()
+            hwm = hwm_store.get_hwm(hwm_name)
+            if hwm:
```

Comment:
```suggestion
            if hwm and hwm.value:
```
There is no need to reset HWM all the time, if it already doesn't have a value